syntax = "proto2";

package schema;

service Router {
   rpc OrderRouter(stream ClientOrder) returns (stream ServerResponse) {}
}

message ClientOrder {
  required uint32 order_id = 1;
  required uint32 user_id = 2;
  required uint32 stock_id = 3; 
  required uint32 event_type = 4;
  required uint32 unit_price = 5;
  required uint32 num_shares = 6;
  required int32  order_type = 7;
}

message ServerResponse {
  required uint32 order_id = 1;
  required uint32 status = 2; // order accepted, error, filled, partially filled, expired, canceled
  required uint32 remaining_shares = 3;
  required uint64 timestamp = 4;
}

